src/src/MetricsWrapper.js

```diff
@@ -0,0 +1,170 @@
+  }
+
+  hasLoaded(key) {
+    return this.metrics.get(key).hasData;
+  }
+
+  async tryUpdateMetrics() {
+    let now = new Date().getTime();
+    await this.#tryUpdateMyCapabilityCosts(now);
+    await this.#tryUpdateResourceCounts(now);
+
+    let has_all_data = true;
+    for (let [_, value] of this.metrics) {
+      if (!value.has_data) {
+        has_all_data = false;
+        break;
+      }
+    }
+
+    return has_all_data;
+  }
+
+  async #tryUpdateResourceCounts(now) {
+    let metric = this.metrics.get(MetricsWrapper.ResourceCountsKey);
+    if (!metric.shouldUpdate(now)) {
+      return;
+    }
+
+    let responseResourceCounts =
+      await this.apiClient.getMyCapabilitiesResourceCounts();
+
+    let resourceCountsMap = new Map();
+    responseResourceCounts.forEach((responseResourceCount) => {
+      let capabilityId = responseResourceCount.capabilityId;
+      if (!resourceCountsMap.has(capabilityId)) {
+        resourceCountsMap.set(capabilityId, new Map());
+      }
+
+      responseResourceCount.awsResourceCounts.forEach((resourceCount) => {
+        resourceCountsMap
+          .get(capabilityId)
+          .set(resourceCount.resourceId, resourceCount.resourceCount);
+      });
+    });
+    if (resourceCountsMap.size > 0) {
+      metric.updateCapabilityMap(now, resourceCountsMap);
+      this.metrics.set(MetricsWrapper.ResourceCountsKey, metric);
+    }
+  }
+
+  async #tryUpdateMyCapabilityCosts(now) {
+    let metric = this.metrics.get(MetricsWrapper.CostsKey);
+    if (!metric.shouldUpdate(now)) {
+      return;
+    }
+
+    let responseCosts = await this.apiClient.getMyCapabilitiesCosts();
+    let costsMap = new Map();
+    responseCosts.forEach((responseCost) => {
+      let capabilityId = responseCost.capabilityId;
+      if (!costsMap.has(capabilityId)) {
+        costsMap.set(capabilityId, []);
+      }
+
+      responseCost.costs.forEach((cost) => {
+        let chartStructure = {
+          name: cost.timeStamp,
+          pv: Math.floor(cost.value),
+        };
+        costsMap.get(capabilityId).push(chartStructure);
+      });
+    });
+    if (costsMap.size > 0) {
+      metric.updateCapabilityMap(now, costsMap);
+      this.metrics.set(MetricsWrapper.CostsKey, metric);
+    }
+  }
+
+  getAwsResourcesTotalCountForCapability(capabilityId) {
+    const metric = this.metrics.get(MetricsWrapper.ResourceCountsKey);
+
+    const counts = metric.capabilitiesMap.get(capabilityId);
+    if (counts === undefined) {
+      return 0;
+    }
+    let total = 0;
+    for (let [_, value] of counts) {
+      total += value;
+    }
+    return total;
+  }
+
+  getAwsResourceCountsForCapability(capabilityId) {
+    const metric = this.metrics.get(MetricsWrapper.ResourceCountsKey);
+    const counts = metric.capabilitiesMap.get(capabilityId);
+    if (counts === undefined) {
+      return new Map();
+    }
+    return counts;
+  }
+
+  getAwsResourceCountForCapability(capabilityId, resourceId) {
```

Comment:
Can we have multiple counts of the same resourceId?
I may misunderstand what is meant by resourceId in this case

Reply:
Here is an example of resource ids and counts:
![image](https://github.com/dfds/selfservice-portal/assets/5738476/7b54ea51-76fe-4833-bdc7-c577f86ccfff)

A real example for using this function could be: "I would like to get the count of databases for this capability". I believe that corresponding Id is: `aws_rds_db_total`